examples/model_selection/plot_roc.py

```diff
@@ -15,118 +15,214 @@
+    y_onehot_test[:, class_id],
+    y_score[:, class_id],
+    name=f"ROC curve for {target_names[class_id]} vs the rest",
+    color="darkorange",
+)
+plt.plot([0, 1], [0, 1], "k--", label="ROC curve for chance level")
+plt.xlim([0.0, 1.0])
+plt.ylim([0.0, 1.05])
+plt.xlabel("False Positive Rate")
+plt.ylabel("True Positive Rate")
+plt.title("One-vs-Rest Receiver operating characteristic example")
+plt.legend()
+plt.show()
 
 # %%
-# Plot of a ROC curve for a specific class
-plt.figure()
-lw = 2
-plt.plot(
-    fpr[2],
-    tpr[2],
+# ROC curve using the OvR micro-average
+# -------------------------------------
+#
+# The micro-average aggregates the contributions from all the classes (using
+# `.ravel`) to compute the average metric. In a multi-class classification
+# setup, micro-average is preferable if you suspect there might be class
+# imbalance.
+
+RocCurveDisplay.from_predictions(
+    y_onehot_test.ravel(),
+    y_score.ravel(),
+    name="One-vs-Rest ROC micro-averaged",
     color="darkorange",
-    lw=lw,
-    label="ROC curve (area = %0.2f)" % roc_auc[2],
 )
-plt.plot([0, 1], [0, 1], color="navy", lw=lw, linestyle="--")
+plt.plot([0, 1], [0, 1], "k--", label="ROC curve for chance level")
 plt.xlim([0.0, 1.0])
 plt.ylim([0.0, 1.05])
 plt.xlabel("False Positive Rate")
 plt.ylabel("True Positive Rate")
-plt.title("Receiver operating characteristic example")
-plt.legend(loc="lower right")
+plt.title("Receiver operating characteristic (micro-averaged)")
+plt.legend()
 plt.show()
 
+# %%
+# In the case where the main interest is not the plot but the ROC-AUC score
+# itself, we can reproduce the value shown in the plot using
+# :class:`~sklearn.metrics.roc_auc_score`.
+
+from sklearn.metrics import roc_auc_score
+
+micro_roc_auc_ovr = roc_auc_score(
+    y_onehot_test,
+    y_score,
+    average="micro",
+)
+
+print(f"Micro-averaged One-vs-Rest ROC AUC score:\n{micro_roc_auc_ovr:.2f}")
 
 # %%
-# Plot ROC curves for the multiclass problem
-# ..........................................
-# Compute macro-average ROC curve and ROC area
+# This is equivalent to computing the ROC curve with
+# :class:`~sklearn.metrics.roc_curve` and then the area under the curve with
+# :class:`~sklearn.metrics.auc` for the raveled classes.
 
-# First aggregate all false positive rates
-all_fpr = np.unique(np.concatenate([fpr[i] for i in range(n_classes)]))
+from sklearn.metrics import roc_curve, auc
+
+fpr = dict()
+tpr = dict()
+roc_auc = dict()
+
+# Compute micro-average ROC curve and ROC area
+fpr["micro"], tpr["micro"], _ = roc_curve(y_onehot_test.ravel(), y_score.ravel())
+roc_auc["micro"] = auc(fpr["micro"], tpr["micro"])
+
+print(f"Micro-averaged One-vs-Rest ROC AUC score:\n{roc_auc['micro']:.2f}")
+
+# %%
+# Notice that by default, the computation of the ROC curve adds a single point
+# at the maximal false positive rate by using linear interpolation and `McClish
+# correction <https://pubmed.ncbi.nlm.nih.gov/2668680/>`_.
```

Comment:
For consistency (see #21088), do you mind using the `:doi:` directive?
The DOI is 10.1177/0272989x8900900307
(https://doi.org/10.1177/0272989x8900900307)